File: py/LC_1.py
class Solution(object):
    def twoSum(self, nums, target):
        cache = dict()

        for i, v in enumerate(nums):
            if v in cache:
                return [cache[v], i]
            cache[target - v] = i
        

def test():
    s = Solution()

    assert s.twoSum([2,7,11,15], 9) == [0, 1]
    assert s.twoSum([3,2,4], 6) == [1, 2]
    assert s.twoSum([3,3], 6) == [0, 1]


def main():
    test()


if __name__ == "__main__":
    main()

File: golang/LC_1.go
package main

import (
	//"fmt"
)

func twoSumBrute(nums []int, target int) []int {

    for i := 0; i < len(nums) - 1; i++ {
		for j := i + 1; j < len(nums); j++ {
			if nums[i] + nums[j] == target {
				return []int{i, j}
			}
		}
	}
	return []int{}
}


func twoSum(nums []int, target int) []int {
	return []int{}
}

/*
func main() {
	fmt.Println(twoSumBrute([]int{2,7,11,15}, 9))
	fmt.Println(twoSumBrute([]int{3,2,4}, 6))
}
*/